lib/rules/no-namespace-imports.js

```diff
@@ -0,0 +1,83 @@
+const {docsUrl} = require('../utilities');
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent namespace import declarations.',
+      category: 'Stylistic Issues',
+      recommended: false,
+      uri: docsUrl('no-namespace-imports'),
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allow: {
+            type: 'array',
+            items: {type: 'string'},
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+    messages: {
+      namespaceImport: `The namespace import declaration for {{name}} should be converted to a default or named import.`,
+    },
+    fixable: 'code',
+  },
+  create(context) {
+    const options = context.options[0] || {};
+    const allowed = options.allow || [];
+
+    function report(node) {
+      const name = node.source.value;
+
+      context.report({
+        node,
+        messageId: 'namespaceImport',
+        data: {name},
+        fix: (fixer) => {
```

Comment:
I don't think it's safe to provide a fixer, we have no idea what members they reference, and turning it into a default import is not going to be correct in many cases.